### PR TITLE
Extend the demo page for the BitDataGrid component (#2357)

### LINKED ISSUE #2357: Extend the demo page for the `BitDataGrid` component
Extend the demo page of the `BitDataGrid` component base on [Microsoft.AspNetCore.QuickGrid](https://aspnet.github.io/quickgridsamples/)

#### Samples to be ported
- [Typical usage](https://aspnet.github.io/quickgridsamples/sample)
- [Virtualizing](https://aspnet.github.io/quickgridsamples/virtualizing)
- [Styling](https://aspnet.github.io/quickgridsamples/styling)

### PR DESCRIPTION
**Pull request by MeysamBahadori**
This closes #2357

### CODE REVIEW

File: src/BlazorUI/Playground/Bit.BlazorUI.Playground/Web/Pages/Components/DataGrid/BitDataGridDemo.razor.cs [hunk 1]
```diff
@@ -10,21 +10,572 @@ namespace Bit.BlazorUI.Playground.Web.Pages.Components.DataGrid;
...
+.grid ::deep tr {
+  height: 1.8rem;
+}
+
+.grid ::deep th:nth-child(1) {
+  width: 15rem;
+}
+
+.grid ::deep th:nth-child(1) .col-options-button {
+  background-image: url('data:image/svg+xml;utf8,<svg xmlns=""http://www.w3.org/2000/svg"" class=""h-6 w-6"" fill=""none"" viewBox=""0 0 24 24"" stroke=""currentColor"" stroke-width=""2""> <path stroke-linecap=""round"" stroke-linejoin=""round"" d=""M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"" /> </svg>');
```
Comment (msynk) [inline src/BlazorUI/Playground/Bit.BlazorUI.Playground/Web/Pages/Components/DataGrid/BitDataGridDemo.razor.cs:309]: please update this part based on the updated content of the actual .scss file

File: src/BlazorUI/Playground/Bit.BlazorUI.Playground/Web/Pages/Components/DataGrid/BitDataGridDemo.razor.cs [hunk 2]
```diff
@@ -10,21 +10,572 @@ namespace Bit.BlazorUI.Playground.Web.Pages.Components.DataGrid;
...
+}
+</style>
+
+<div class=""grid"">
+    <BitDataGrid Items = ""@FilteredItems"" ResizableColumns=""true"" Pagination=""@pagination"">
+        <BitDataGridPropertyColumn Property = ""@(c => c.Name)"" Sortable=""true"" Class=""column1"">
+            <ColumnOptions>
+                <input type = ""search"" autofocus @bind = ""nameFilter"" @bind:event=""oninput"" />
```
Comment (msynk) [inline src/BlazorUI/Playground/Bit.BlazorUI.Playground/Web/Pages/Components/DataGrid/BitDataGridDemo.razor.cs:317]: please update this part based on the updated content of the actual .razor file

File: src/BlazorUI/Playground/Bit.BlazorUI.Playground/Web/Pages/Components/DataGrid/BitDataGridDemo.razor.cs [hunk 3]
```diff
@@ -10,21 +10,572 @@ namespace Bit.BlazorUI.Playground.Web.Pages.Components.DataGrid;
...
+    new Country { Code = ""BS"", Name = ""Bahamas"", Medals = new Medals { Gold = 2, Silver = 0, Bronze = 0 } },
+    new Country { Code = ""BH"", Name = ""Bahrain"", Medals = new Medals { Gold = 0, Silver = 1, Bronze = 0 } },
+    ...
+};
+";
 
     BitDataGridPaginationState pagination = new() { ItemsPerPage = 15 };
     IQueryable<Country> items;
-    string nameFilter = string.Empty;
+    IQueryable<Country> data;
```
Comment (msynk) [inline src/BlazorUI/Playground/Bit.BlazorUI.Playground/Web/Pages/Components/DataGrid/BitDataGridDemo.razor.cs:566]: please use a proper name for this variable

**Review by msynk — CHANGES_REQUESTED** [review]

**Comment by ysmoradi** [comment]
The virtualization sample needs to retrieve data from the remote data source

File: src/BlazorUI/Playground/Bit.BlazorUI.Playground/Web/Pages/Components/DataGrid/BitDataGridDemo.razor.cs [hunk 4]
```diff
@@ -564,18 +599,41 @@ private async Task<BitDataGridItemsProviderResult<Country>> Get7CountriesAsync()
...
     protected override async Task OnInitializedAsync()
     {
         items = (await GetCountriesAsync(0, null, null, true, CancellationToken.None)).Items.AsQueryable();
         data = (await Get7CountriesAsync()).Items.AsQueryable();
+        foodRecallData= (await GetfoodRecallAsync(0, null, null, true,virtualSampleNameFilter, CancellationToken.None)).Items.AsQueryable();
+    }
 
+    private async Task<BitDataGridItemsProviderResult<FoodRecall>> GetfoodRecallAsync(int startIndex, int? count, string sortBy, bool sortAscending,string search, CancellationToken cancellationToken)
+    {
+        var url = NavManager.GetUriWithQueryParameters("http://api.fda.gov/food/enforcement.json", new Dictionary<string, object?>
```
Comment (ysmoradi) [inline src/BlazorUI/Playground/Bit.BlazorUI.Playground/Web/Pages/Components/DataGrid/BitDataGridDemo.razor.cs:590]: Use https://

File: src/BlazorUI/Playground/Bit.BlazorUI.Playground/Web/Pages/Components/DataGrid/BitDataGridDemo.razor.cs [hunk 5]
```diff
@@ -1,15 +1,28 @@
...
 using Bit.BlazorUI.Playground.Web.Pages.Components.ComponentDemoBase;
+using Microsoft.AspNetCore.Components;
 
 namespace Bit.BlazorUI.Playground.Web.Pages.Components.DataGrid;
 
 public partial class BitDataGridDemo
 {
+    public BitDataGridDemo()
+    {
+        ServicePointManager.ServerCertificateValidationCallback = delegate (object sender, X509Certificate certificate, X509Chain chain, SslPolicyErrors sslPolicyErrors)
```
Comment (ysmoradi) [inline src/BlazorUI/Playground/Bit.BlazorUI.Playground/Web/Pages/Components/DataGrid/BitDataGridDemo.razor.cs:21]: ?

File: src/BlazorUI/Playground/Bit.BlazorUI.Playground/Web/Pages/Components/DataGrid/BitDataGridDemo.razor.cs [hunk 6]
```diff
@@ -564,18 +599,41 @@ private async Task<BitDataGridItemsProviderResult<Country>> Get7CountriesAsync()
...
+                { "skip", startIndex },
+                { "limit", count },
+                { "search", search },
+            });
+
+        using var httpResponse = await Http.GetAsync(url, cancellationToken);       
+
+        if (!httpResponse.IsSuccessStatusCode)
+        {
+            return new BitDataGridItemsProviderResult<FoodRecall>() { Items=new List<FoodRecall>() { } , TotalItemCount=0 };
```
Comment (ysmoradi) [inline src/BlazorUI/Playground/Bit.BlazorUI.Playground/Web/Pages/Components/DataGrid/BitDataGridDemo.razor.cs:601]: Format the code please

Comment (ysmoradi) [inline src/BlazorUI/Playground/Bit.BlazorUI.Playground/Web/Pages/Components/DataGrid/BitDataGridDemo.razor.cs:601]: Use System.Text.Json Source Generator to improve the performance of this (and the rest of http calls if any)

**Review by ysmoradi — CHANGES_REQUESTED** [review]

**Review by MeysamBahadori — COMMENTED** [review]
Ready for review

**Comment by MeysamBahadori** [comment]
Ready for review

**Comment by MeysamBahadori** [comment]
plz review again @ysmoradi

**Review by msynk — APPROVED** [review]

**Pull request merged**